dask/dataframe/expr.py

```diff
@@ -0,0 +1,163 @@
+import operator
+
+from toolz import first
+
+from .. import threaded
+from ..base import is_dask_collection, tokenize
+from ..blockwise import blockwise
+from ..expr import Expr
+from ..utils import funcname
+from .core import finalize
+
+
+class ExprFrame(Expr):
```

Comment:
Feel free to ignore this thought, since I am still wrapping my mind around the process here.

Is there a clean way that`ExprFrame` could either inherit attributes from `_Frame`, or that `_Frame` could be retrofitted to inherit from `Expr`?  It seems like the easiest way to gradually introduce the `Expr`-subclass approach would be to allow something like `TimeSeries().map_partitions(<some-func>)` without **requiring** a `ExprFrame.map_partitions` method.  I realize that we would probably *want* to implement a specialized subclass anyway - I am just using `map_partitions` as an example of a common Dask-DataFrame operation.

Reply:
I hope so :)

In general I think that you're right to look for a way to gradually introduce this idea into the codebase.  If we can find a gradual implementation then we can avoid a lot of pain.